FP_layout/app/src/main/java/com/example/dhyatmika/fp_layout/DriverActivity.java

```diff
@@ -3,101 +3,259 @@
 import android.content.SharedPreferences;
+import android.hardware.Sensor;
+import android.hardware.SensorEvent;
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.util.Log;
 import android.view.View;
-import android.widget.Toast;
+import android.widget.Button;
+import android.widget.TextView;
 
-import com.android.volley.AuthFailureError;
 import com.android.volley.NetworkResponse;
 import com.android.volley.Request;
-import com.android.volley.RequestQueue;
-import com.android.volley.Response;
 import com.android.volley.VolleyError;
-import com.android.volley.toolbox.JsonObjectRequest;
-import com.android.volley.toolbox.StringRequest;
-import com.android.volley.toolbox.Volley;
-
 import org.json.JSONException;
 import org.json.JSONObject;
 
-import java.sql.Driver;
 import java.util.HashMap;
-import java.util.Map;
 
-public class DriverActivity extends AppCompatActivity {
+public class DriverActivity extends AppCompatActivity implements SensorEventListener{
+
+    private Context getContext() {
+        return DriverActivity.this.getApplicationContext();
+    }
+
+    private SharedPreferences getPreference() {
+        return Helper.getPreference(this.getContext());
+    }
+
+    private String getToken() {
+        return Helper.getToken(this.getContext());
+    }
+
+    private SensorManager mSensorManager;
+
+    private Sensor accelerometer;
+
+    private boolean switcher = true;
+    private boolean startStopSwitcher = true;
+    private boolean firstPress = true;
+    private boolean change = true;
+
+    private Context mContext;
+
+
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_driver);
 
-        SharedPreferences preference = getSharedPreferences("LOGIN_CREDENTIALS", Context.MODE_PRIVATE);
-        if (preference.getString("token", "").equals("")) {
+        mContext = getApplicationContext();
+        String token = this.getToken();
+        if (token.equals("")) {
             Intent intent = new Intent(this, LoginActivity.class);
             startActivity(intent);
         }
+
+        checkTokenValidity();
+
+        mSensorManager = (SensorManager) getSystemService(Context.SENSOR_SERVICE);
+
+        if(accelerometer == null){
+            //do something
+        }
     }
 
-    public void logout(View view) {
+    private void checkTokenValidity() {
 
-        // get token for logging out
-        SharedPreferences preference = getSharedPreferences("LOGIN_CREDENTIALS", Context.MODE_PRIVATE);
-        String token = preference.getString("token", "");
+        // get token
+        String token = this.getToken();
+        String url = AppConfig.getValidate(token);
 
+        Helper.MakeJsonObjectRequest(mContext, Request.Method.GET, url, null, new VolleyResponseCallback() {
```

Comment:
<subjective>
I'm not quite sure about this `MakeJsonObjectRequest` method on the `Helper` class, it looks like over-engineering from my end.

The other method on the helper class, is something I agree on being really helpful, but for this particular instance, I feel iffy about it. Let me think a bit before going back here.
</subjective>